src/main/java/blockchain/LightChainNode.java

```diff
@@ -700,6 +706,34 @@ public boolean isSound(Transaction t) {
 		}
 	}
 
+	
+		public boolean isCorrect(Transaction t, Contract c) throws IOException { 
```

Comment:
`isCorrect` is specific to the contract you have written, so it should not be part of the business logic of `LightChainNode`. To preserve the object-oriented principle, I suggest implementing a `CorrectnessVerifier` class, that is passed to the `LightChainNode` constructor. The `CorrectnessVerifier` class is an abstract class, and has an abstract function called `isCorrect`. Then depending on the value of `params.getChain()` either the LightChain native one or the contract-based one is getting passed to the `LightChain` node. So essentially the code would be something like this (this is a pseudocode and may need some hammering):

```
                         LightChainNode n; 
                         CorrectnessVerifier cv; 
			if (params.getChain() == false) { 
                                    // LightChainCV extends CorrectnessVerifier for native LightChain
                                     cv = new LightChainCV();
                           }
			else{	
                                     // ContractCV extends CorrectnessVerifier for the contract mode
                                     cv = new ContractCV(c);
                         }
                         LightChainNode n = new LightChainNode(cv); 
```

And then you replace line 637-638 above with `cv.isCorrect(t);` so the `LightChainNode` remains completely _dumb_ about whether it is executed in _contract_ mode or _native_ mode. 

Also, the function is missing JavaDoc.